docs/content/design/proxy.md

```diff
@@ -62,3 +62,25 @@ Private module filters are string globs that tell the Athens proxy what is a pri
 ### Exclude Lists for Public Modules
 
 Exclude lists for public modules are also globs that tell the Athens proxy what modules it should never download from any upstream proxy. For example, the string `github.com/arschles/**` tells the Athens proxy to always return `404 Not Found` to clients.
+
+## Catalog Endpoint
+
+The proxy provides a `/catalog` service endpoint to fetch all the modules and version contained in the local storage.
+The endpoint accepts a continuation token and a page size parameter in order to provide paginated results.
+
+A query is of the form
+
+`https://proxyurl/catalog?token=foo&limit=47`
+
+Where token is an optional continuation token and limit is the desired size of the returned page.
+
+
+The result is a json with the following structure:
+
+```
+{"modules": [{"Module":"github.com/athens-artifacts/no-tags","Version":"v1.0.0"}],
```

Comment:
Can you also make module and version lower cased here:
https://github.com/gomods/athens/commit/2142d9989c13bcff86984c2a71a8725547a10ba5